dvc/repo/checkout.py

```diff
@@ -15,7 +15,14 @@ def _cleanup_unused_links(repo):
         for out in stage.outs
         if out.scheme == "local"
     ]
-    repo.state.remove_unused_links(used)
+
+    unused = repo.state.get_unused_links(used)
+    for link in unused:
+        logger.info(
+            "Removing '{}' as it already exists in the current worktree.", link
```

Comment:
cc @jorgeorpinel - please review the messages in this PR

Reply:
I couldn't think of succinct messages. :cry:

Reply:
The message is not very useful for `info`, IMHO. How about we  move it to `debug`?

Reply:
Also, it is pretty weird to split old `remove_unused_links` just to print a message here. Or am I missing something?

Reply:
> Also, it is pretty weird to split old `remove_unused_links` just to print a message here. Or am I missing something?

yes: https://github.com/iterative/dvc/pull/3297#issuecomment-584204079

Reply:
> First of all, I tried to maintain consistency: state does not have logger.info anywhere. Therefore, I did not try to introduce one, because I wanted to separate concerns (ui vs state/model) wherever possible, and it seemed like a good idea here for doing just that. slightly_smiling_face

So you moved it just to not use `logger.info`? That is a pretty weak argument for doing that, tbh :slightly_smiling_face: Please see https://github.com/iterative/dvc/pull/3297#discussion_r377620466 , I think that it should be like that or we should keep `remove_unused_links` in one piece. Let me know what you think.

Reply:
I don't consider separation of UI and business logic to be a weak argument. But probably as it's a cli app and already done this way on most of the places, maybe it is.

>  I think that it should be like that or we should keep remove_unused_links in one piece. Let me know what you think.

I am fine, either way. Previously, I did not want to separate those too far off like you suggested below and tried to just get the stuff done. :see_no_evil:
It does not really need to be separated if `logger.info` call is kept on `dvc.state` as it's only used in a single place.

Reply:
> cc @jorgeorpinel - please review the messages in this PR

I see #3326 replaced this and only has a debug string change